flutter/lib/src/native/platform/transport.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:ffi';
+import 'package:ffi/ffi.dart';
+import 'package:web_socket_channel/io.dart';
+
+import 'transport.g.dart' as bridge;
+
+/// Registers functions allowing Rust to operate Dart [IOWebSocketChannel]s.
+void registerFunctions(DynamicLibrary dl) {
+  bridge.registerFunction(
+    dl,
+    connect: Pointer.fromFunction(_connect),
+    listen: Pointer.fromFunction(_listen),
+    send: Pointer.fromFunction(_send),
+    close: Pointer.fromFunction(_close),
+  );
+}
+
+/// Connects to the provided [addr] and returns [IOWebSocketChannel] for it.
+Object _connect(Pointer<Utf8> addr) {
+  return IOWebSocketChannel.connect(Uri.parse(addr.toDartString()));
+}
+
+/// Subscribes on [IOWebSocketChannel.stream] with provided [onMessage] and
+/// [onClose] callbacks.
+void _listen(IOWebSocketChannel ws, Function onMessage, Function onClose) {
+  ws.stream.listen((msg) {
+    if (msg is String) {
+      onMessage(msg);
+    }
+  }, onDone: () {
+    onClose();
+  });
+}
+
+/// Sends the provided [message] to the provided [IOWebSocketChannel].
+void _send(IOWebSocketChannel ws, Pointer<Utf8> message) {
```

Comment:
Also, should we use `IOWebSocketChannel` which is just a wrapper over [`dart:io.WebSocket`](https://api.dart.dev/stable/dart-io/WebSocket-class.html)? Wouldn't be using raw `WebSocket` sufficient enough without requiring third-party deps?